source/max.js

```diff
@@ -18,5 +18,23 @@ import _curry2 from './internal/_curry2.js';
  *      R.max(789, 123); //=> 789
  *      R.max('a', 'b'); //=> 'b'
  */
-var max = _curry2(function max(a, b) { return b > a ? b : a; });
+var max = _curry2(function max(a, b) {
+  if (a === b) { return b; }
+
+  function safeMax(x, y) {
+    if ((x > y) !== (y > x)) { return y > x ? y : x; }
+    return undefined;
+  }
+
+  const maxByValue = safeMax(a, b);
+  if (maxByValue !== undefined) { return maxByValue; }
+
+  const maxByType = safeMax(typeof a, typeof b);
+  if (maxByType !== undefined) { return maxByType === typeof a ? a : b; }
+
+  const maxByStringValue = safeMax(toString(a), toString(b));
+  if (maxByStringValue !== undefined) { return maxByStringValue === toString(a) ? a : b; }
```

Comment:
I don't like recalculating `toString(a)` here.  Perhaps we can save `stringA` and `stringB`